google-cloud-clients/google-cloud-datastore/src/test/java/com/google/cloud/datastore/DatastoreTest.java

```diff
@@ -691,6 +691,27 @@ public void testRunKeyQueryWithOffset() {
     assertEquals(2, numberOfEntities);
   }
 
+  @Test
+  public void testRunKeyQueryWithLimit() {
+    Query<Key> query = Query.newKeyQueryBuilder().setLimit(0).build();
+    QueryResults results = datastore.run(query);
+    int count = 0;
+    while (results.hasNext()) {
+      results.next();
+      count++;
+    }
+    assertEquals(0, count);
+    datastore.put(ENTITY1, ENTITY2);
+    Query<Key> keyQuery = Query.newKeyQueryBuilder().setLimit(2).build();
+    QueryResults queryResults = datastore.run(keyQuery);
+    int resultCount = 0;
+    while (queryResults.hasNext()) {
+      queryResults.next();
+      resultCount++;
+    }
+    assertEquals(2, resultCount);
+  }
```

Comment:
Does below works? If yes then can you please update it

```java
@Test
  public void testRunKeyQueryWithLimit() {
    datastore.put(ENTITY1, ENTITY2);
    Query<Key> keyQuery = Query.newKeyQueryBuilder().setLimit(2).build();
    QueryResults queryResults = datastore.run(keyQuery);
    int resultCount = 0;
    while (queryResults.hasNext()) {
      queryResults.next();
      resultCount++;
    }
    assertEquals(2, resultCount);
   Query<Key> query = Query.newKeyQueryBuilder().setLimit(0).build();
    QueryResults results = datastore.run(query);
    int count = 0;
    while (results.hasNext()) {
      results.next();
      count++;
    }
    assertEquals(0, count);
  }
```

reason being even though there are entities in datastore after setting limit to 0 u r getting 0 as results.

Reply:
done. working fine even if we change sequence of code